internal/serverinstall/ecs.go

```diff
@@ -0,0 +1,2177 @@
+		if *r.ResourceType == "AWS::ECS::TaskDefinition" {
+			_, err := ecsSvc.DeregisterTaskDefinition(&ecs.DeregisterTaskDefinitionInput{
+				TaskDefinition: r.ResourceArn,
+			})
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// InstallRunner implements Installer.
+func (i *ECSInstaller) InstallRunner(
+	ctx context.Context,
+	opts *InstallRunnerOpts,
+) error {
+	ui := opts.UI
+	log := opts.Log
+
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: i.config.Region,
+		Logger: log,
+	})
+	if err != nil {
+		return err
+	}
+
+	var (
+		logGroup      string
+		executionRole string
+		runSvcArn     *string
+	)
+	lf := &Lifecycle{
+		Init: func(ui terminal.UI) error {
+			sess, err = utils.GetSession(&utils.SessionConfig{
+				Region: i.config.Region,
+				Logger: log,
+			})
+			if err != nil {
+				return err
+			}
+			executionRole, err = i.SetupExecutionRole(ctx, ui, log, sess)
+			if err != nil {
+				return err
+			}
+
+			logGroup, err = i.SetupLogs(ctx, ui, log, sess, defaultRunnerLogGroup)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		Run: func(ui terminal.UI) error {
+			runSvcArn, err = i.LaunchRunner(
+				ctx, ui, log, sess,
+				opts.AdvertiseClient.Env(),
+				executionRole,
+				logGroup,
+			)
+			return err
+		},
+
+		Cleanup: func(ui terminal.UI) error { return nil },
+	}
+
+	if err := lf.Execute(log, ui); err != nil {
+		return err
+	}
+
+	log.Debug("runner service started", "arn", *runSvcArn)
+
+	return nil
+}
+
+var (
+	serverResourceQuery = "{\"ResourceTypeFilters\":[\"AWS::AllSupported\"],\"TagFilters\":[{\"Key\":\"%s\",\"Values\":[]}]}"
+	runnerResourceQuery = "{\"ResourceTypeFilters\":[\"AWS::AllSupported\"],\"TagFilters\":[{\"Key\":\"%s\",\"Values\":[\"%s\"]}]}"
+)
+
+func (i *ECSInstaller) UninstallRunner(
+	ctx context.Context,
+	opts *InstallOpts,
+) error {
+	ui := opts.UI
+	log := opts.Log
+
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Uninstalling Runner resources...")
+	defer func() { s.Abort() }()
+
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: i.config.Region,
+		Logger: log,
+	})
+	if err != nil {
+		return err
+	}
+	rgSvc := resourcegroups.New(sess)
+
+	query := fmt.Sprintf(runnerResourceQuery, defaultRunnerTagName, defaultRunnerTagValue)
+	results, err := rgSvc.SearchResources(&resourcegroups.SearchResourcesInput{
+		ResourceQuery: &resourcegroups.ResourceQuery{
+			Type:  aws.String(resourcegroups.QueryTypeTagFilters10),
+			Query: aws.String(query),
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	resources := results.ResourceIdentifiers
+	var clusterArn string
+	for _, r := range resources {
+		if *r.ResourceType == "AWS::ECS::Cluster" {
+			clusterArn = *r.ResourceArn
+		}
+	}
+	s.Update("Deleting ECS resources...")
+	if err := deleteEcsCommonResources(ctx, sess, clusterArn, resources); err != nil {
+		return err
+	}
+	s.Done()
+	s = sg.Add("Deleting Cloud Watch Log Group resources...")
```

Comment:
Maybe same comment here, could combine these into an Update and finish with `"Deleted ECS resources"`

Reply:
Agreed, done in 6b212967